src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp

```diff
@@ -802,10 +803,33 @@ class spell_gothik_shadow_bolt_volley : public SpellScript
     }
 };
 
+class at_gothik_entrance : public OnlyOnceAreaTriggerScript
+{
+public:
+    at_gothik_entrance() : OnlyOnceAreaTriggerScript("at_gothik_entrance") { }
+
+    bool _OnTrigger(Player* player, AreaTrigger const* /*at*/) override
+    {
+        if (InstanceScript* instance = player->GetInstanceScript())
+        {
+            if (instance->GetBossState(BOSS_HORSEMAN) == DONE)
+                return false;
+
+            // Do not continue! Turn back while there's still time!
+            if (Creature* zeliek = instance->GetCreature(DATA_SIR_ZELIEK))
+                zeliek->AI()->Talk(SAY_ZELIEK_INTRO_2, 30s);
+
+            return true;
+        }
+        return false;
+    }
+};
+
 void AddSC_boss_gothik()
 {
     new boss_gothik();
     new npc_boss_gothik_minion();
     new npc_gothik_trigger();
     RegisterSpellScript(spell_gothik_shadow_bolt_volley);
+    new at_gothik_entrance();
```

Comment:
Can new register method be used here?

Reply:
Register macros for Areatriggers do not exist yet.

[I implemented something like this, but out of scope for this PR](https://github.com/azerothcore/azerothcore-wotlk/commit/fad198a445c6d91d4f368afffab679b48ffbb204#diff-6396c45ad9207c72c30ee43ea53d6bc708bb7dc4a7a2b5e3414b4a301abcbccd)
I 
```
struct at_deathknight_wing_entrance : OnlyOnceAreaTriggerScript
{
    PrepareOnlyOnceAreaTriggerScript(at_deathknight_wing_entrance);

    bool _OnTrigger(Player* player, AreaTrigger const* /*at*/) override
    {
        ...
    }
};
```

versus

```
class at_deathknight_wing_entrance : public OnlyOnceAreaTriggerScript
{
public:
    at_deathknight_wing_entrance() : OnlyOnceAreaTriggerScript("at_deathknight_wing_entrance") { }

    bool _OnTrigger(Player* player, AreaTrigger const* /*at*/) override
    {
        ...
    }
};
```